packages/editor-ui/src/plugins/codemirror/completions/__tests__/completions.test.ts

```diff
@@ -311,10 +301,10 @@ describe('Resolution-based completions', () => {
 	describe('bracket-aware completions', () => {
 		const { $input } = mockProxy;
 
-		test('should return bracket-aware completions for: {{ $input.item.json.str.|() }}', () => {
-			vi.spyOn(workflowHelpers, 'resolveParameter').mockReturnValue($input.item.json.str);
+		test('should return bracket-aware completions for: {{ $input.item.json?.str.|() }}', () => {
+			vi.spyOn(workflowHelpers, 'resolveParameter').mockReturnValue($input.item?.json.str);
 
-			const found = completions('{{ $input.item.json.str.|() }}');
+			const found = completions('{{ $input.item.json?.str.|() }}');
```

Comment:
have you considered updating the type for `$input` instead? 

are not these completions tests for how the code/expression editor is used? the type completions there don't require `?` and most users don't think about that. I am concerned we are losing important test cases here.

Reply:
The optional chaining inside the expression was a mistake from search/replacing 😓, fixed now. The type of `$input` is from `WorkflowDataProxy` and it doesn't know if `$input.item.json` will exist, I don't think it's possible to fix without a type cast, so I'd rather keep the optional chaining.

Reply:
ah that makes more sense. since it's strings.